src/Model/Linter.php

```diff
@@ -1,19 +1,40 @@
 <?php
 
-namespace DTL\BehatLint\Model;
+namespace DTL\GherkinLint\Model;
 
 use Cucumber\Gherkin\GherkinParser;
+use Generator;
 
 class Linter
 {
-    public function __construct(private GherkinParser $parser)
-    {
+    public function __construct(
+        private GherkinParser $parser,
+        /**
+         * @var Rule[]
+         */
+        private array $rules
+    ) {
     }
 
-    public function lint(string $uri, string $contents): FeatureDiagnostics
+    /**
+     * @return Generator<FeatureDiagnostic>
+     */
+    public function lint(string $uri, string $contents): Generator
     {
-        $node = $this->parser->parseString($uri, $contents);
+        $envelopes = iterator_to_array($this->parser->parseString($uri, $contents));
+
+        foreach ($envelopes as $envelope) {
```

Comment:
@ciaranmcnulty not really sure why there are multiple envelopes? is it expected to pass `includeSource`, `includeGherkinDocument`, `includePickles` (what are Pickles?) to the parser to determine what is returned?

Reply:
Messages are used for more than just gherkin, it's also the test execution events (see https://github.com/cucumber/common/tree/main/messages#message-overview) - ignore any that aren't relevant

Just for clarity though here's what the parser can emit:

 * Source - this is the raw `.feature` file and its URL. You would probably only want this if you want to output a chunk of the file in output? It's more convenient than recreating from the AST
 * GherkinDocument - this is the AST, you probably want this
 * Pickle - this is the AST normalised into something like `list<list<Step>>` - it's for a test runner to not have to worry about things like Background. You aren't likely to want this

Reply:
(basically envelope is so we can naively decode JSON without knowing out-of-band what 'type' it is, then look at its fields to see what it was)

Reply:
ok so could "filter" envelopes by checking if `gherkinDocument` is NULL or not I guess